superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -772,10 +828,20 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
   };
 
   const handleBackButtonOnConnect = () => {
-    if (editNewDb) {
-      setHasConnectedDb(false);
-    }
+    if (editNewDb) setHasConnectedDb(false);
+    if (importingModal) setImportingModal(false);
```

Comment:
another nit that we can prob address in a later cleanup. The name `importingModal` sounds like a current state where the modal is importing.